components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java

```diff
@@ -92,6 +93,297 @@
+        if (CollectionUtils.isNotEmpty(parseAttemptForV2.getMessages())) {
+            log.debug("Errors found when parsing OAS definition");
+        }
+        return parseAttemptForV2.getSwagger();
+    }
+
+    /**
+     * This method returns the boolean value which checks whether the swagge is included default security scheme or not
+     *
+     * @param swaggerContent resource json
+     * @return is default is given already
+     * @throws APIManagementException
+     */
+    private boolean isDefaultGiven(String swaggerContent) throws APIManagementException {
+        Swagger swagger = getSwagger(swaggerContent);
+        boolean isDefaultIsGiven = true;
+        Map<String, SecuritySchemeDefinition> securityDefinitions = swagger.getSecurityDefinitions();
+        OAuth2Definition checkDefault = (OAuth2Definition) securityDefinitions.get(SWAGGER_SECURITY_SCHEMA_KEY);
+        if (checkDefault == null) {
+            isDefaultIsGiven = false;
+        }
+        return isDefaultIsGiven;
+    }
+
+    /**
+     * This method returns the oauth scopes according to the given swagger(version 2)
+     *
+     * @param swaggerContent resource json
+     * @return scope set as all defaults
+     * @throws APIManagementException
+     */
+    @Override
+    public Set<Scope> injectOtherScopesToDefaultScheme(String swaggerContent) throws APIManagementException {
+        Swagger swagger = getSwagger(swaggerContent);
+
+        Map<String, SecuritySchemeDefinition> securityDefinitions = swagger.getSecurityDefinitions();
+        OAuth2Definition oAuth2Definition;
+        List<String> otherSetOfSchemes = new ArrayList<>();
+        Set<Scope> scopeSet = new HashSet<>();
+        boolean isDefaultAvailable = isDefaultGiven(swaggerContent);
+
+        if (securityDefinitions != null && !isDefaultAvailable) {
+            //If there is no default type schemes set a one
+            OAuth2Definition newDefault = new OAuth2Definition();
+            securityDefinitions.put(SWAGGER_SECURITY_SCHEMA_KEY, newDefault);
+            for (Map.Entry<String, SecuritySchemeDefinition> definition : securityDefinitions.entrySet()) {
+                if (definition.getKey() != SWAGGER_SECURITY_SCHEMA_KEY) {
+                    otherSetOfSchemes.add(definition.getKey());
+                    //Check for default one
+                    OAuth2Definition noneDefaultFlowType = (OAuth2Definition) definition.getValue();
+                    OAuth2Definition defaultTypeFlow = (OAuth2Definition) securityDefinitions.get(SWAGGER_SECURITY_SCHEMA_KEY);
+                    Map<String, String> noneDefaultFlowScopes = noneDefaultFlowType.getScopes();
+                    Map<String, String> defaultTypeScopes = defaultTypeFlow.getScopes();
+                    if (defaultTypeScopes == null) {
+                        defaultTypeScopes = new HashMap<>();
+                    }
+                    for (Map.Entry<String, String> input : noneDefaultFlowScopes.entrySet()) {
```

Comment:
Doesn't this give NPEs during above situations like api-key security present where scopes are not there?